### PR TITLE
Feature algorithms enhancement - edit


### PR DESCRIPTION
**Pull request by vbo-acc**
Lossless data compression algorithm. 
The idea is to assign variable-length codes to input characters, lengths of the assigned codes are based on the frequencies of corresponding characters.

### CODE REVIEW

File: src/main/java/NodeCompress.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+public class NodeCompress {
+    int item;
+    char c;
+    NodeCompress left;
+    NodeCompress right;
+}
+
+// For comparing the nodes
+class ImplementComparator implements Comparator<NodeCompress> {
+    public int compare(NodeCompress x, NodeCompress y) {
```
Comment (vbozh) [inline src/main/java/NodeCompress.java:15]: please, add tests to proceed merge pull request

**Review by vbozh — CHANGES_REQUESTED** [review]

File: src/main/java/NodeCompress.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+public class NodeCompress {
+    int item;
+    char c;
+    NodeCompress left;
+    NodeCompress right;
+}
+
+// For comparing the nodes
+class ImplementComparator implements Comparator<NodeCompress> {
```
Comment (vbozh) [inline src/main/java/NodeCompress.java:14]: please, add tests to proceed merge pull request

**Review by vbozh — CHANGES_REQUESTED** [review]

File: src/main/java/NodeCompress.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
+// Huffman Coding in Java
```
Comment (CK-qa) [inline src/main/java/NodeCompress.java:1]: Please follow general Code Style rules described in the Guide.md

File: src/main/java/NodeCompress.java (hunk 2 above)
Comment (vbozh) [inline src/main/java/NodeCompress.java:14]: @vbo-acc please take a look.

**Comment by vbozh** [comment]
**fff**

File: src/main/java/canal/CleanUpMe.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+public class CleanUpMe {
+
+    int foo;
+
+    public String[] process(String[] value) {
+        return value;
+    }
+
+    void bar() {
+        this.foo += 1;
```
Comment (vbozh) [inline src/main/java/canal/CleanUpMe.java:12]: increment

**Comment by vbozh** [comment]
test and delete

File: src/main/java/canal/CodeAnalyzeMe.java [hunk 5]
```diff
@@ -1,10 +1,38 @@
...
+
+    public CodeAnalyzeMe() {
+    }
+
     void method() {
+        System.out.println();
     }
 
+    //label change
+
```
Comment (vbozh) [inline src/main/java/canal/CodeAnalyzeMe.java:19]: single comment

File: src/main/java/canal/CodeAnalyzeMe.java [hunk 6]
```diff
@@ -1,10 +1,38 @@
...
+    String s;
+    Objects objects;
+    Integer i;
+
+    public CodeAnalyzeMe() {
+    }
+
     void method() {
+        System.out.println();
     }
```
Comment (vbozh) [inline src/main/java/canal/CodeAnalyzeMe.java:16]: start review with a comment

**Review by vbozh — CHANGES_REQUESTED** [review]
request changes

**Pull request merged**